src/rules/prefer-importing-jest-globals.ts

```diff
@@ -0,0 +1,186 @@
+          fix(fixer) {
+            const sourceCode = context.getSourceCode();
+            const [firstNode] = sourceCode.ast.body;
+
+            let firstNodeValue;
+
+            if (firstNode.type === 'ExpressionStatement') {
+              const firstExpression = firstNode.expression as Literal;
+              const { value } = firstExpression;
+
+              firstNodeValue = value;
+            }
+
+            const useStrictDirectiveExists =
+              firstNode.type === 'ExpressionStatement' &&
+              firstNodeValue === 'use strict';
+
+            if (useStrictDirectiveExists) {
+              return fixer.insertTextAfter(
+                firstNode,
+                `\n${createFixerImports(isModule, jestFunctionsToImport)}`,
+              );
+            }
+
+            const importNode = sourceCode.ast.body.find(
+              node =>
+                node.type === AST_NODE_TYPES.ImportDeclaration &&
+                node.source.value === '@jest/globals',
+            );
+
+            if (
+              importNode &&
+              importNode.type === AST_NODE_TYPES.ImportDeclaration
+            ) {
+              const existingImports = importNode.specifiers.map(specifier => {
+                /* istanbul ignore else */
+                if (specifier.type === 'ImportSpecifier') {
+                  return specifier.imported?.name;
+                }
+
+                // istanbul ignore next
+                return null;
+              });
+              const allImports = [
+                ...new Set([
+                  ...existingImports.filter(
+                    (imp): imp is string => imp !== null,
+                  ),
+                  ...jestFunctionsToImport,
+                ]),
+              ];
+
+              return fixer.replaceText(
+                importNode,
+                createFixerImports(isModule, allImports),
+              );
+            }
+
+            const requireNode = sourceCode.ast.body.find(
+              node =>
+                node.type === 'VariableDeclaration' &&
+                node.declarations.some(
+                  declaration =>
+                    declaration.init &&
+                    (declaration.init as any).callee &&
+                    (declaration.init as any).callee.name === 'require' &&
+                    (declaration.init as any).arguments?.[0]?.type ===
+                      'Literal' &&
+                    (declaration.init as any).arguments?.[0]?.value ===
+                      '@jest/globals',
+                ),
+            );
+
+            if (requireNode && requireNode.type === 'VariableDeclaration') {
+              const existingImports =
+                requireNode.declarations[0]?.id.type === 'ObjectPattern'
+                  ? requireNode.declarations[0]?.id.properties?.map(
+                      property => {
+                        /* istanbul ignore else */
```

Comment:
lots of coverage ignores here - can we add tests which cover these cases? If not - why do we have the uncovered branches to begin with?

Reply:
See the [current thread](https://github.com/jest-community/eslint-plugin-jest/pull/1490#discussion_r1478408327) about code coverage. Maybe you know some tricks?